drivers/test-fw/src/bin/csrng_fail_repcnt_soc_threshold_tests.rs

```diff
@@ -0,0 +1,51 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    csrng_fail_repcnt_soc_threshold_tests.rs
+
+Abstract:
+    https://opentitan.org/book/hw/ip/entropy_src/doc/theory_of_operation.html#repetition-count-test
+
+    File contains test cases for CSRNG API when the physical entropy source
+    has a stuck bit on at least one of the four external RNG wires.
+
+    We expect the Repetition Count health check to fail for these tests.
+
+    The thresholds are set in SoC registers.
+--*/
+#![no_std]
+#![no_main]
+
+use caliptra_drivers::Csrng;
+use caliptra_error::CaliptraError;
+use caliptra_registers::{csrng::CsrngReg, entropy_src::EntropySrcReg, soc_ifc::SocIfcReg};
+use caliptra_test_harness::test_suite;
+
+fn test_boot_fail_repcnt_check() {
+    let csrng_reg = unsafe { CsrngReg::new() };
+    let entropy_src_reg = unsafe { EntropySrcReg::new() };
+    let mut soc_ifc_reg = unsafe { SocIfcReg::new() };
+    soc_ifc_reg
+        .regs_mut()
+        .cptra_i_trng_entropy_config_1()
+        .write(|w| w.repetition_count(20));
```

Comment:
In all these tests, do you mind setting these from the SoC side rather than the uC? If there's a problem in the hardware I'd like these tests to notice.